### ISSUE TITLE
Produce DX Package

### ISSUE THREAD

**Issue by afawcett** [afawcett/force-di#1]
Create a package [per this](https://andyinthecloud.com/2018/06/16/salesforce-dx-packages-and-open-source/) to ease usage of this library.

**Comment by PeterLinBiz** [comment]
I think we have a few options to do this.
Three of the possible options might be as below.
Option A: (One master branch for both #1 and #12)
Assuming we have #12 done already and all classes have the "di_" prefix at master branch.
Then when we package DX (Unlocked) Package (package using the master branch), if we use "di" as namespace, then Injector.cls will be di__di_Injector.cls.
Pros: Only one master branch to maintain compared to Option B.
Cons: "di__di_" prefix looks not so perfect.

Option B: (One master branch for "di_" prefix #12, another branch (maybe call "unlocked_package" branch) for no-prefix for unlock package #1)
Assuming we have #12 done already and all classes have the "di_" prefix at a TO-BE-MARGE branch.
Then we might want to "git branch unlocked_package" when we are on master branch to preserve a no-prefix branch.
Then on master branch we might "git merge TO-BE-MARGE".
Pros: When we publish and upgrade unlocked package versions, we will have a nice looking di__Injector.cls from the unlocked_package branch.
Cons: All enhancements needs to have two sets, one (pull request) for the master branch ("di_" prefix) another (pull request) for the unlocked_package branch.

Option C: (One master branch for "di_" prefix for both #1 and #12)
Assuming we have #12 done already and all classes have the "di_" prefix at master branch.
When we publish Unlocked Package(package using the master branch), if we use "di" as namespace. Every time, we need to preform refactoring to remove "di_" prefix, then Injector.cls will be di__Injector.cls.
Pros: When we publish and upgrade unlocked package version, we will have a nice looking di__Injector.cls from the master branch (after refactoring to remove "di_" prefix). Only one master branch to maintain compared to Option B.
Cons: Every time, we need to preform refactoring to remove "di_" prefix for publishing and upgrading unlocked package versions.

**Comment by PeterLinBiz** [comment]
Personally, I prefer Option A to maintain only one master branch with #12 completed ("di_Injector.cls") and probably "DInj__di_Injector.cls" in the unlocked package.

**Comment by PeterLinBiz** [comment]
I also noticed that [based on this](https://andyinthecloud.com/2018/06/16/salesforce-dx-packages-and-open-source/) as above, we need to consider **global access modifier** when we prepare to publish unlocked package. So will Option A branching strategy still workable?

**Comment by ImJohnMDaniel** [comment]
I am with @douglascayers on this one.  I would prefer to see the option in #12 where we add a `dl_` prefix to all metadata.

That will allow people the ease of easily incorporating the code base into either their own enterprise-org unlocked packages or ISV's managed packages.  

I believe the maintenance around supporting a managed package for this project is not trivial over time and will cause more issues than it is worth.

Basically, I am in favor of closing this ticket with no action and let #12 proceed.

**Comment by douglascayers** [comment]
We now provide unlocked package for easy install as well as instructions how to install manually from GitHub if desired.

https://github.com/afawcett/force-di#other-resources-on-dependency-injection-design-pattern
